model/project_ref_test.go

```diff
@@ -2071,7 +2071,50 @@ func TestGithubPermissionGroups(t *testing.T) {
 		require.NoError(err)
 		assert.Equal(orgGroup[0].Name, intersection.Name)
 
-		// Fields that were set on orgGroup[0].
+		assert.True(intersection.HasNoPermissions())
+	})
+
+	t.Run("Intersection of two no permissions should return no permissions", func(t *testing.T) {
```

Comment:
Can we add a test where the reason it has no permissions is because the intersection has no overlap? 
For example, to check if something like this: 

> 
```
 {
       checks: read
 }
```
  and 
```
  {
        pull_request: read
  } 
```
>     

Is considered to have no permission?